app/src/main/java/com/example/playlistmaker/settings/di/SettingsDomainModule.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.playlistmaker.settings.di
+
+import com.example.playlistmaker.Util.App
+import com.example.playlistmaker.settings.data.impl.SettingsRepositoryImpl
+import com.example.playlistmaker.settings.domain.api.SettingsInteractor
+import com.example.playlistmaker.settings.domain.api.SettingsRepository
+import com.example.playlistmaker.settings.domain.impl.SettingsInteractorImpl
+import org.koin.dsl.module
+
+val settingsDomainModule = module {
+    single<App> { App() }
+
+    factory<SettingsRepository> {
+        SettingsRepositoryImpl(switchPreferences = get(), app = get())
+    }
```

Comment:
🍏 Конутруктор для App обычно вызывает система, поэтому нам не требуется это делать. Koin имеет доступ к экземпляру нашего application, плюс при инициализации Koin ты передаешь ему ссылку на application. Поэтому достаточно сделать так:
```suggestion
    factory<SettingsRepository> {
        SettingsRepositoryImpl(switchPreferences = get(), app = androidApplication() as App)
    }
```